File: practica_final/GameShop-1/GameShop/app/src/main/java/com/example/t11navigation/adapters/AdaptadorCarrito.java
package com.example.t11navigation.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.t11navigation.ModelJuego;
import com.example.t11navigation.R;
import com.example.t11navigation.fragments.Carrito;

import java.util.ArrayList;

public class AdaptadorCarrito extends RecyclerView.Adapter<AdaptadorCarrito.myviewholder> {

    ArrayList<Carrito> carrito;

    public AdaptadorCarrito( ArrayList<Carrito> carrito) {
        this.carrito = carrito;
    }

    @NonNull
    @Override
    public myviewholder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.lista_carrito, parent, false);
        return new myviewholder(view);

    }

    @Override
    public void onBindViewHolder(@NonNull myviewholder holder, int position) {
        holder.titulo.setText(carrito.get(position).getTituloJuego());
        holder.precio.setText(carrito.get(position).getPrecioJuego());

    }

    @Override
    public int getItemCount() {
        return carrito.size();
    }

    class myviewholder extends RecyclerView.ViewHolder {
        TextView titulo;
        TextView precio;

        public myviewholder(@NonNull View itemView) {
            super(itemView);
            titulo = itemView.findViewById(R.id.txt_titulo);
            precio = itemView.findViewById(R.id.txt_precio);

        }

    }
}

File: practica_final/practica_final/app/src/main/java/com/example/practica_final/adapter/AdaptadorCarrito.kt
package com.example.practica_final.adapter

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.practica_final.FragmentDetalle
import com.example.practica_final.SecondActivity
import com.example.practica_final.R
import com.example.practica_final.model.Producto

class AdaptadorCarrito(var listaProductos: ArrayList<Producto>, var context: Context) : RecyclerView.Adapter<AdaptadorCarrito.MyHolder>() {

    class MyHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        // template --> aspecto del XML
        var nombreTextView: TextView
        var precioTextView: TextView
        var imagenImageView: ImageView

        init {
            //Localiza los views del layout
            nombreTextView = itemView.findViewById(R.id.nombre_carrito_producto)
            precioTextView = itemView.findViewById(R.id.precio_carrito_producto)
            imagenImageView = itemView.findViewById(R.id.imagen_carrito_producto)
        }

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyHolder {
        // crea la plantilla y la retorna, necesitando el XML
        val view: View = LayoutInflater.from(context).inflate(R.layout.carrito_recycler_layout,parent,
            false)
        return MyHolder(view)
    }

    override fun getItemCount(): Int {
        // tamaño de la lista de elementos a mostrar <Artista>
        return listaProductos.size
    }

    override fun onBindViewHolder(holder: MyHolder, position: Int) {
        // enlaza la plantilla con el elemento de la posicion
        val dato = listaProductos[position]
        holder.nombreTextView.text =  dato.nombre
        holder.precioTextView.text = dato.precio.toString()
        Glide.with(context).load(dato.imagen)
            .into(holder.imagenImageView)

    }

}